server/punishments.ts

```diff
@@ -1122,10 +1122,13 @@ export const Punishments = new class {
 			user.resetName();
 			user.updateIdentity();
 		} else {
-			if (punishUserid === '#hostfilter') {
-				user.popup(`Due to spam, you can't chat using a proxy. (Your IP ${user.latestIp} appears to be a proxy.)`);
-			} else if (punishUserid === '#ipban') {
-				user.popup(`Your IP (${user.latestIp}) is not allowed to chat on PS, because it has been used to spam, hack, or otherwise attack our server. Make sure you are not using any proxies to connect to PS.`);
+			if (punishUserid === '#hostfilter' || punishUserid === '#ipban') {
+				if (user.latestHostType !== 'proxy') {
+					// Disabled their proxy but haven't logged out.
+					user.send(`|popup||html|You are currently locked due to being on a proxy. If you have disabled it, log out fully by typing <code>/logout</code> in any chatroom.`);
```

Comment:
I think you could put the logout button in this popup, if that'd be more helpful?

Reply:
I tried [these](https://hastebin.com/dijeziviro.cs) and it seems a client change is needed.  I see that [this](https://github.com/smogon/Pokemon-Showdown-Client/commit/52973f240affcbf49bcb1d296f1511f96809d4fc) exists for Preact.  If you can get it to work, feel free to suggest how.